lib/jsdom/level2/html.js

```diff
@@ -1886,14 +1927,15 @@ function loadFrame (frame) {
 
   var src = frame.src.trim() === '' ? 'about:blank' : frame.src;
   var parentDoc = frame._ownerDocument;
-
+  
   // If the URL can't be resolved or the src attribute is missing / blank,
   // then url should be set to the string "about:blank".
   // (http://www.whatwg.org/specs/web-apps/current-work/#the-iframe-element)
   var url = resourceLoader.resolveResourceUrl(parentDoc, src);
+  var urlObj = url.match(/(.+?):(.*)/);
```

Comment:
Use utils's URL constructor and check the resulting protocol, instead of using regexes.